### PR TITLE
ENG-3610 show proper 422 error

### PR DESCRIPTION
**Pull request by ichalagashvili**

### CODE REVIEW

File: src/state/component-repository/components/actions.js [hunk 1]
```diff
@@ -298,7 +298,7 @@ const procceedWithInstall = (component, body, resolve, dispatch, logProgress, lo
           const versionUnavailable = data && data.message;
           if (versionUnavailable) {
             dispatch(addErrors([data.message]));
-            dispatch(addToast(data.message, TOAST_ERROR));
+            dispatch(addToast(data.message || { id: 'componentRepository.components.genericError' }, TOAST_ERROR));
```
Comment (Dv92) [inline src/state/component-repository/components/actions.js]: I think we should save `{ id: 'componentRepository.components.genericError' }` inside a `const DEFAULT_ERROR_MESSAGE` for centralize this value and don't replicate it.

Comment (chalagashvili) [reply src/state/component-repository/components/actions.js]: done!

Comment (ichalagashvili) [reply src/state/component-repository/components/actions.js]: done!

File: src/state/component-repository/components/actions.js [hunk 2]
```diff
@@ -326,7 +326,10 @@ export const installECRComponent = (component, version, logProgress, resolvedIns
         postECRComponentInstallPlan(component, version)
           .then((response) => {
             response.json().then(({ payload: installPlan }) => {
-              if (!installPlan.hasConflicts) {
+              if (!installPlan) {
+                dispatch(addToast({ id: 'componentRepository.components.genericError' }, TOAST_ERROR));
```
Comment (Dv92) [inline src/state/component-repository/components/actions.js]: you have lost this 🙂

File: src/state/component-repository/components/actions.js [hunk 3]
```diff
@@ -368,7 +371,7 @@ export const getInstallPlan = component => dispatch => (
             dispatch(toggleConflictsModal(true, installPlan, component, null, true));
           }
         } catch (e) {
-          dispatch(addToast(e.message, TOAST_ERROR));
+          dispatch(addToast(e.message || { id: 'componentRepository.components.genericError' }, TOAST_ERROR));
```
Comment (Dv92) [inline src/state/component-repository/components/actions.js]: also this 🙂

**Review by Dv92 — APPROVED** [review]

**Pull request merged**